Express/Typescript template fails to start

### What version of Remix are you using?

1.15.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

`npx create-remix@latest`, select express adapter, select Typescript, run `npm run dev` 

### Expected Behavior

The dev server to startup

### Actual Behavior

```
[nodemon] starting `cross-env NODE_ENV=production node ./server.js --require dotenv/config ./server.js`
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module '/.../my-remix-app/server.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.15.0
[nodemon] app crashed - waiting for file changes before starting...
```

for the record, the typecheck script also fails right the install

```
npm run typecheck

> typecheck
> tsc

node_modules/@remix-run/node/dist/upload/fileUploadHandler.d.ts:47:22 - error TS2420: Class 'NodeOnDiskFile' incorrectly implements interface 'File'.
  Property 'prototype' is missing in type 'NodeOnDiskFile' but required in type 'File'.

47 export declare class NodeOnDiskFile implements File {
                        ~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2572:5
    2572     prototype: Blob;
             ~~~~~~~~~
    'prototype' is declared here.


Found 1 error in node_modules/@remix-run/node/dist/upload/fileUploadHandler.d.ts:47
```

--- a/templates/express/server.js
+++ b/templates/express/server.js
@@ -1,11 +1,13 @@
-import path from "path";
-import express from "express";
-import compression from "compression";
-import morgan from "morgan";
-import { createRequestHandler } from "@remix-run/express";
-import { installGlobals } from "@remix-run/node";
+const path = require("path");
+const express = require("express");
+const compression = require("compression");
+const morgan = require("morgan");
+const { createRequestHandler } = require("@remix-run/express");
+const { installGlobals } = require("@remix-run/node");
 
 installGlobals();
+
+const BUILD_DIR = path.join(process.cwd(), "build");
 
 const app = express();
 
@@ -26,33 +28,34 @@ app.use(express.static("public", { maxAge: "1h" }));
 
 app.use(morgan("tiny"));
 
-const BUILD_DIR = path.join(process.cwd(), "build");
-const MODE = process.env.NODE_ENV;
 app.all(
   "*",
-  MODE === "production"
-    ? createRequestHandler({ build: require(BUILD_DIR), mode: MODE })
-    : (...args) => {
+  process.env.NODE_ENV === "development"
+    ? (req, res, next) => {
         purgeRequireCache();
 
         return createRequestHandler({
           build: require(BUILD_DIR),
-          mode: MODE,
-        })(...args);
+          mode: process.env.NODE_ENV,
+        })(req, res, next);
       }
+    : createRequestHandler({
+        build: require(BUILD_DIR),
+        mode: process.env.NODE_ENV,
+      })
 );
-
 const port = process.env.PORT || 3000;
+
 app.listen(port, () => {
-  console.log(`✅ Express server listening on port ${port}`);
+  console.log(`Express server listening on port ${port}`);
 });
 
 function purgeRequireCache() {
   // purge require cache on requests for "server side HMR" this won't let
   // you have in-memory objects between requests in development,
   // alternatively you can set up nodemon/pm2-dev to restart the server on
-  // file changes, we prefer the DX of this though, so we've included it
-  // for you by default
+  // file changes, but then you'll have to reconnect to databases/etc on each
+  // change. We prefer the DX of this, so we've included it for you by default
   for (const key in require.cache) {
     if (key.startsWith(BUILD_DIR)) {
       delete require.cache[key];

